platforms/common/include/px4_platform_common/atomic.h

```diff
@@ -58,7 +58,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-#if defined(__PX4_NUTTX)
+#if defined(__PX4_NUTTX) && (defined (CONFIG_BUILD_FLAT) || defined (__KERNEL__))
```

Comment:
Turn this into a single define somewhere?

Reply:
Sure, it will get prettier. Can you suggest a location ? Maybe px4_impl_os.cmake ?

Reply:
I was more concerned with accidental mistakes slipping in.

Yes somewhere in px4_os_add_flags seems fine.

https://github.com/PX4/PX4-Autopilot/blob/92fbd86b463dabd4c1ff0e4cec5c4205dec402f5/platforms/nuttx/cmake/px4_impl_os.cmake#L73

Reply:
Thanks, I'll figure it out. It makes sense to turn this into a singular flag to remove confusion and as you said, avoid accidents

Reply:
Resolving as outdated